src/GazeboUnderwater.cpp

```diff
@@ -77,16 +78,15 @@ namespace gazebo_underwater
     physics::Inertial GazeboUnderwater::computeModelInertial(physics::ModelPtr model) const
     {
         Inertial inertial(0);
-        inertial.SetMOI(math::Matrix3::ZERO);
-        Inertial temp;
+        inertial.SetMOI(GzMatrix3(Matrix3d::Zero));
         physics::Link_V links = model->GetLinks();
         for (physics::Link_V::iterator it = links.begin(); it != links.end(); ++it)
             if(!(*it)->GetKinematic())
             {
                 // Set Inertial's CoG related with the parent link
-                temp = *(*it)->GetInertial();
-                math::Pose pose = (*it)->GetRelativePose();
-                pose.pos += pose.rot.RotateVector(temp.GetCoG());
+                Inertial temp = *(*it)->GetInertial();
+                Pose3d pose = GzGetIgn((**it), RelativePose, ());
+                pose.CoordPositionAdd(pose.Rot().RotateVector(GzGetIgn(temp, CoG, ())));
```

Comment:
The `CoordPositionAdd` does not do the same [thing](https://bitbucket.org/osrf/gazebo/src/a29d8dabed9a14fb6bafdcd31c84b160c00b127c/gazebo/math/Pose.cc?at=gazebo7_7.0.0&fileviewer=file-view-default#Pose.cc-164) as the previous.